### PR TITLE
Tilemap

### PR DESCRIPTION
**Pull request by cedric-h**

### CODE REVIEW

File: src/tilemap/farmable.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub struct Farmable {}
```
Comment (cedric-h) [inline src/tilemap/farmable.rs:1]: The idiomatic syntax for making a ZST like this is
`pub struct Farmable;`

Furthermore, I'm not sure whether or not this should really be in the `tilemap` module; Isn't farm behavior more generic?

Comment (cedric-h) [reply src/tilemap/farmable.rs:1]: Or, not necessarily more generic but certainly better suited to a farm module.

File: src/tilemap/mod.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    kind: graphics::AppearanceKind::image("dirt"), //tile_details.image.clone()
+                    alignment: graphics::Alignment::Center,
+                    z_offset: -1000.0,
+                    ..Default::default()
+                })
+                .add(Iso2::translation(
+                    (TILE_SIZE / 2 + TILE_SIZE * x) as f32,
+                    (TILE_SIZE / 2 + TILE_SIZE * y) as f32,
+                ))
+                .add(Tile {});
```
Comment (cedric-h) [inline src/tilemap/mod.rs:35]: Why is it important or why would it be useful in the future to mark an entity as a Tile?
Can't we offer a less specific API through the ECS?

File: config.toml [hunk 3]
```diff
@@ -7,3 +7,32 @@ keyframes = [
...
+
+tilemap = """\
+gr,gr,gr,gr,gr
+di,di,di,di,di
+gr,di,gr,di,gr
+di,gr,di,gr,di
+gr,gr,di,di,gr
+gr,gr,di,di,gr
+gr,gr,gr,gr,gr
+fr,fr,fr,fr,bd
```
Comment (cedric-h) [inline config.toml:19]: In most monospaced text renderers, the ratio of the length of a character to the height of a character is 1:2.
For this reason, if we have two letters per map tile, you can have a square for each tile, making the text tile map a scale model of the graphical one.

File: src/tilemap/mod.rs (hunk 2 above)
Comment (cedric-h) [reply src/tilemap/mod.rs:35]: Even if we do somehow find some reason to mark something as a tile in the future, that may never be the case and until then, this component is just bloat.

File: src/tilemap/mod.rs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+pub use self::farmable::Farmable;
+use crate::config::TileProperty;
+use crate::phys::collision;
+use crate::{graphics, Iso2, Vec2};
+use hecs::{EntityBuilder, World};
+use ncollide2d::shape::Cuboid;
+use std::collections::HashMap;
+
+const TILE_SIZE: usize = 16;
```
Comment (cedric-h) [inline src/tilemap/mod.rs:11]: Consider adding a `const HALF_TILE: usize = TILE_SIZE / 2;` constant.

File: src/tilemap/mod.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                ))
+                .add(Tile {});
+
+            if tile_details.farmable {
+                tile_builder.add(Farmable {});
+            }
+            if tile_details.collidable {
+                tile_builder
+                    .add(collision::CollisionStatic)
+                    .add(Cuboid::new(Vec2::new(
```
Comment (cedric-h) [inline src/tilemap/mod.rs:43]: Consider using `Vec2::repeat(TILE_SIZE as f32 / 2.0)` or `Vec2::repeat(HALF_TILE as f32)` here.

File: src/tilemap/mod.rs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    let mut tile_builder = EntityBuilder::new();
+    let default = TileProperty::default();
+
+    tilemap.split_whitespace().enumerate().for_each(|(y, row)| {
+        row.split(',').enumerate().for_each(|(x, tile)| {
+            let tile_details = tile_prop.get(tile).unwrap_or(&default);
+
+            tile_builder
+                .add(graphics::Appearance {
+                    kind: graphics::AppearanceKind::image("dirt"), //tile_details.image.clone()
```
Comment (cedric-h) [inline src/tilemap/mod.rs:26]: Why is the appearance hard coded in?

File: src/tilemap/mod.rs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+use crate::{graphics, Iso2, Vec2};
+use hecs::{EntityBuilder, World};
+use ncollide2d::shape::Cuboid;
+use std::collections::HashMap;
+
+const TILE_SIZE: usize = 16;
+
+pub struct Tile {}
+
+//static\image
```
Comment (cedric-h) [inline src/tilemap/mod.rs:15]: This comment serves no purpose?

**Review by cedric-h — COMMENTED** [review]
You've done well!

File: src/tilemap/mod.rs (hunk 6 above)
Comment (RybekU) [reply src/tilemap/mod.rs:26]: Oopsie

**Pull request closed without merging**